### DISCUSSION TITLE
Get value from another field in nested where clause

### DISCUSSION THREAD

**Discussion by paulwer** [prisma/prisma#24661] [Q&A] [answered]
### Question

Hi everyone, 
we are looking for a posibility to perform a nested where clause with a value from the level above.

We found this feature / thread: https://github.com/prisma/prisma/discussions/15392
But it seems, that this only provided the ability to compare values within one table / level.

Does anyone have a suggestion for us?

We want to do domething like this:
```
    const liveMaintenancePackages = await this.prisma.liveMaintenancePackage.findMany({
            where: {
                liveMachine: {
                    deleted: { equals: null },
                    contracts: {
                        some: {
                            timeline: {
                                none: {
                                    liveMaintenancePackage: { id: { equals: this.prisma.liveMaintenancePackage.fields.id } }  // compare the id with the id of the result
                                }
                            },
                        },
                    },
                },
        });
```

**Comment by ludralph** [comment]
Hi @paulwer 👋 

If I understand correctly, you want to perform a nested where clause that involves comparing a value from a parent level to a nested level. Unfortunately, Prisma currently does not support directly comparing values across different levels in a nested query. This feature is limited to comparisons within the same table or model. Can you open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase?

As a workaround, you can achieve your goal by splitting the query into multiple steps. For example:

- Fetch the parent data first: Retrieve the `liveMaintenancePackage` data you need.
- Use the fetched data to filter the nested query: Use the fetched `id` to filter the nested contracts.

```js
// Step 1: Fetch the liveMaintenancePackage data
const liveMaintenancePackages = await this.prisma.liveMaintenancePackage.findMany({
  where: {
    // Your conditions for liveMaintenancePackage
  },
});

// Step 2: Use the fetched data to filter the nested query
const filteredPackages = await Promise.all(
  liveMaintenancePackages.map(async (package) => {
    const contracts = await this.prisma.contract.findMany({
      where: {
        liveMachine: {
          deleted: null,
        },
        timeline: {
          none: {
            liveMaintenancePackageId: package.id,
          },
        },
      },
    });

    return {
      ...package,
      contracts,
    };
  })
);

// Filter out packages with no contracts
const result = filteredPackages.filter((package) => package.contracts.length > 0);

console.log(result);
```

**Comment by paulwer** [comment, marked as answer]
Hi @ludralph,
thanks for your answer, we have created an Feature Request accordingly.
I also wrote about our current workaound to use a view, because multi level fethcing would not work considdering our requirement to fetch large amount of entries.

https://github.com/prisma/prisma/issues/24682
